azurerm/internal/services/netapp/netapp_volume_resource.go

```diff
@@ -496,3 +794,19 @@ func flattenNetAppVolumeMountIPAddresses(input *[]netapp.MountTargetProperties)
 
 	return results
 }
+
+func flattenNetAppVolumeDataProtectionReplication(input *netapp.VolumePropertiesDataProtection) []interface{} {
+	if input == nil || input.Replication == nil || strings.ToLower(string(input.Replication.EndpointType)) != "dst" {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"endpoint_type":             string(input.Replication.EndpointType),
```

Comment:
we will need to nil check Replication here

Reply:
The following check is done in the beginning of the function and I think it covers null check:

````go
	if input == nil || input.Replication == nil || strings.ToLower(string(input.Replication.EndpointType)) != "dst" {
		return []interface{}{}
	}
```

it checks the possible path, input then if replication is nil then if endpoint type is != "dst" wich nil would be caught as well.